### PR TITLE
Fix bug where dotfiles are found in package root #73

### LINKED ISSUE #73: Dotfile found in package root
## Environment 
On MacOS
```
brew install lima
limactl start default
```

Run build in container with lima:
```
nerdctl run --rm -it -v \
          (pwd)/debbuild:/results \
          --entrypoint bash ghcr.io/prebuiltmpr/mpr-builder:focal
```

Version:
```
builder@277920aa859f:/results$ makedeb --version
makedeb 8.5.9-1
Alpha Release
Installed from APT
```
## Problem
Can't not find dot files, but no dot files inside `$pkgdir`:
```
make[3]: Nothing to be done for 'install-exec-am'.
 /usr/bin/mkdir -p '/results/libmxf/pkg/libmxf/usr/lib/pkgconfig'
 /usr/bin/install -c -m 644 libMXF-1.0.pc '/results/libmxf/pkg/libmxf/usr/lib/pkgconfig'
make[3]: Leaving directory '/results/libmxf/src/libmxf'
make[2]: Leaving directory '/results/libmxf/src/libmxf'
make[1]: Leaving directory '/results/libmxf/src/libmxf'
[#] Tidying install...
  [->] Removing libtool files...
  [->] Purging unwanted files...
  [->] Removing static library files...
  [->] Stripping unneeded symbols from binaries and libraries...
  [->] Compressing man and info pages...
[#] Checking for packaging issues...
[!] Dotfile found in package root '/results/libmxf/pkg/libmxf/.*'
builder@277920aa859f:/results/libmxf$ ls -A
PKGBUILD  libmxf  pkg  src
builder@277920aa859f:/results/libmxf$ cd pkg/libmxf/
builder@277920aa859f:/results/libmxf/pkg/libmxf$ ls -A
usr
builder@277920aa859f:/results/libmxf/pkg/libmxf$
```

My `PKGBUILD` file:
```
# Maintainer: 

pkgname=libmxf
_commit=9b9860d
pkgver=1.0.4.$_commit
pkgrel=1
pkgdesc="libMXF is a low-level C library for reading and writing the SMPTE ST 377-1 MXF file format."
arch=('x86_64')
url="https://sourceforge.net/p/bmxlib/libmxf/ci/master/tree/"
license=('BSD 3-clause license')
makedepends=('uuid-dev' 'git' 'fakeroot' 'make' 'automake' 'autoconf' 'libtool' 'build-essential')
provides=('libmxf')
conflicts=('libmxf')
source=("git+https://git.code.sf.net/p/bmxlib/libmxf.git")
sha256sums=('SKIP')

prepare() {
    cd $pkgname
    git checkout $_commit
}

build() {
    cd $pkgname
    bash autogen.sh
    bash ./configure --prefix=/usr --disable-examples CFLAGS="-std=gnu99"
    make -j$(nproc)
}

package() {
    cd $pkgname
    make DESTDIR="$pkgdir" prefix="/usr" install
}


```

### Installation source

makedeb Package Repository (MPR)

### Installed release type

Alpha

### Installed makedeb version

makedeb 8.5.9-1 Alpha Release

### Installed makepkg version

makepkg-template (pacman) 6.0.0

### PR DESCRIPTION
**Pull request by JehhB**
#73

### CODE REVIEW

**Comment by hwittenborn** [comment]
Hey, thanks for the PR! I'm working on some other stuff at the moment, but I'll try to get this merged by tomorrow or Monday.

**Comment by hwittenborn** [comment]
Not sure what's up, but these changes seem to be causing the false positives, I've never personally encountered them before, and it looks like the CI hasn't either.

What environment are you using makedeb in?

**Comment by JehhB** [comment]
I'm using debian unstable

**Comment by knokelmaat** [comment]
I have the same issue, also on debian unstable. Anything we can do on our end to help you debug?

**Comment by knokelmaat** [comment]
I modified the original `dotfiles.sh` to print some directory info:



```
check_dotfiles() {
	local ret=0

	ls "$pkgdir" 			# Extra info
	echo "$pkgdir"/.* 		# Extra info

	for f in "${pkgdir}/.*"; do

    		echo $f 		# Extra info
		echo ${f##*/} 		# Extra info
		echo . 			# Extra info

		[[ ${f##*/} == . || ${f##*/} == .. ]] && continue
		error "$(gettext "Dotfile found in package root '%s'")" "$f"
		ret=1
	done
	return $ret
}

```

This gives the following output:

```
christophe@localhost:~/Downloads/una-bin$ makedeb -si
[#] Making package: una-bin 3.2.0-1 (Tue 06 Sep 2022 10:32:15 AM CEST)
[#] Checking for missing dependencies...
[#] Loading extensions...
[#] Retrieving sources...
  [->] Found una
  [->] Found una-completion
  [->] Found una-updater
  [->] Found una.8.adoc
[#] Validating source files with sha256sums...
    una ... Skipped
    una-completion ... Skipped
    una-updater ... Skipped
    una.8.adoc ... Skipped
[#] Extracting sources...
[#] Removing existing $pkgdir/ directory...
[#] Entering fakeroot environment...
[#] Starting package()...
[#] Checking for packaging issues...
usr
/home/christophe/Downloads/una-bin/pkg/una-bin/.*
/home/christophe/Downloads/una-bin/pkg/una-bin/.*
.*
.
[!] Dotfile found in package root '/home/christophe/Downloads/una-bin/pkg/una-bin/.*
```

**Comment by hwittenborn** [comment]
Sorry for the late reply @JehhB and @knokelmaat, I didn't get anything in my GitHub notifications or I just overlooked it or something :P.

> I have the same issue, also on debian unstable. Anything we can do on our end to help you debug?

Not at the moment, I should be able to get a Debian unstable Docker container going shortly to get this all tested, I think a know a pretty good way to get this solved though.

File: src/functions/lint_package/dotfiles.sh [hunk 1]
```diff
@@ -28,11 +28,10 @@ source "$LIBRARY/util/message.sh"
...
 check_dotfiles() {
-	local ret=0
-	for f in "$pkgdir"/.*; do
-		[[ ${f##*/} == . || ${f##*/} == .. ]] && continue
-		error "$(gettext "Dotfile found in package root '%s'")" "$f"
-		ret=1
-	done
-	return $ret
+    local ret=0
+    for f in $(ls -Ad "$pkgdir"/.*); do
```
Comment (hwittenborn) [inline src/functions/lint_package/dotfiles.sh]: We should be able to just avoid `ls` completely by using something like this. It does add another variable into the function, but it stops all this globbing nonsense that's causing issues:

```sh
local dotfiles
mapfile -t dotfiles < <(find -mindepth 1 -maxdepth 1 | sed 's|^\./||' | grep '^\.')
```

**Review by hwittenborn — CHANGES_REQUESTED** [review]

File: src/functions/lint_package/dotfiles.sh (hunk 1 above)
Comment (JehhB) [reply src/functions/lint_package/dotfiles.sh]: What if I replace line 32 with
```bash
for f in $(find "$pkgdir" -mindepth 1 -maxdepth 1 -name '.*'); do
```
It also eliminate the globing issue and also doesn't introduce any new variable. I think it also better to keep the same error message wherein the full path of the dotfile is displayed.

Comment (hwittenborn) [reply src/functions/lint_package/dotfiles.sh]: If there's a dotfile named something like `.one two`, then that gets printed as two separate files with that approach. It's not the end of the world if that were to happen, but I'd prefer to stick with `mapfile` just so errors all line up properly.

**Comment by hwittenborn** [comment]
Hey I'm gonna go ahead and get this merged, all these MRs have been around for quite a bit and I'm just wanting them closed so they ain't lingering around forever

**Pull request merged**